### DISCUSSION TITLE
Command not found on Yarn 3 PnP workspaces

### DISCUSSION THREAD

**Discussion by johnathanludwig** [microsoft/vscode-discussions#53] [Extension Development QnA] [answered]
I have a typescript extension that works fine, but an error occurs if you use the extension in a workspace that is using Yarn 3 PnP and has the typescript version set to the [workspace version](https://yarnpkg.com/getting-started/editor-sdks#vscode). The user sees a "command not found" error.

When debugging I see the error "Your application tried to access vscode, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound. Required package: vscode." 

Anyone have ideas on what to do to solve it?

**Comment by isidorn** [comment, marked as answer]
No clue. But sounds like you should file a new issue in the VS Code repo? Reproducable steps would help.

**Reply by johnathanludwig** [reply]
I tried to make a reproduction repo but couldn't get it to reproduce. I think it was more of an issue on the yarn side with PNP. To be fair I had reproduced it before in other repos, just hadn't made one specific for reproducing the problem. 

In my original repo where I was primarily having this issue I updated yarn and removed the `.pnp.cjs` and `.pnp.loader.mjs` file so they would be rebuilt and everything seems to be good. It hasn't happened in a few days now. I'm guessing it was a bug fixed by the update or something in the pnp files.

Thanks for looking!

**Comment by jmathew** [comment]
For the record, I have run into this problem with my extension as well. And so far it does only fail when my extension command is run in a project that has yarn PNP. But switching TS back to vscode version does not seem to fix it.
